addons/mail/static/src/core/common/thread.js

```diff
@@ -153,6 +156,18 @@ export class Thread extends Component {
         onMounted(async () => {
             this.lastJumpPresent = this.props.jumpPresent;
             await this.threadService.fetchNewMessages(this.props.thread);
+            if (this.props.thread.id === "inbox") {
+                const [user] = await this.orm.read(
+                    "res.users",
+                    [this.store.user.user?.id],
+                    ["partner_id", "notification_type"]
+                );
+                this.store.Persona.insert({
+                    id: this.store.user.id,
+                    notification_preference: user.notification_type,
+                    type: "partner",
+                });
```

Comment:
Don't need `Persona.insert()` when you already have a model. You can immediately assign the value:
```js
this.store.user.notification_preference = user.notification_type;
```

Also I would call a formatter rather than `orm.read()`, as this leaks the conversion of `notification_type` to `notification_preference`.

Reply:
What is the problem of directly calling from orm and leaking the conversion?

Reply:
The conversion is an internal concern of the server. By leaking it here, it puts more maintenance effort for changes.

For example, if we decide to rename field `res.users/notification_type` to `res.users/notification_preference`, we'd expect to just change the field and the formatter, but these LOCs should also be changed.

This is not probably covered by a test, and even if that's the case we'll waste a lot of time (at least 1 hour for a red runbot) to spot that the rename here is incorrect. Imagine if we leak it in other places... We could probably waste half a day or more just for a mere field rename.

Reply:
If we fetch the data each time we load the thread, there will be a lot of rpc increase. So I changed the implementation. Now, only if the `res.partner` records have been modified, then the model will be updated with no extra rpcs. But in this case, there is no way to format the name of the field since it is from the `change` https://github.com/odoo/odoo/pull/136174/files#diff-691e6a29537279840262a3eb9bdd6d9c512acc1d0f2360f8f321b411a6989e22R50 But concerning the performance, I think it is a better choice.